powerline/segment.go

```diff
@@ -124,7 +146,11 @@ func getGitInformation() (string, bool) {
 	return status, staged
 }
 
-func GitSegment(t Theme) Segment {
+func GitSegment(cwdParts []string, t Theme) Segment {
+	if !isGitDir(cwdParts) {
```

Comment:
Maybe, maybe not. getGitInformation at the moment gets information from git, isGitDir is a pre-filter. cwdParts was the closest information to what was needed that is already available to other segments.